test/test_import_hook.py

```diff
@@ -74,49 +74,133 @@ def _test_import_hook(importhook_tempdir, typechecker):
     assert counter + 1 == jaxtyping._test_import_hook_counter
 
 
-def test_import_hook_typeguard_old(importhook_tempdir):
-    _test_import_hook(importhook_tempdir, ("typeguard", "typechecked"))
+# Tests start here...
 
 
-def test_import_hook_typeguard(importhook_tempdir):
-    _test_import_hook(importhook_tempdir, "typeguard.typechecked")
+def test_import_hook_function_beartype(importhook_tempdir):
+    try:
+        import beartype  # noqa: F401,F811
+    except ImportError:
+        pytest.skip("Beartype not installed")
+    else:
+
+        def f(x, *args, **kwargs):
+            import beartype  # noqa: F401,F811
+
+            return beartype.beartype(x, *args, **kwargs)
```

Comment:
What's up with this wrapping? Just `_test_import_hook(importhook_tempdir, beartype.beartype)` should suffice I think?

Unrelatedly, I've realised the try/except can be cleaned up into a `beartype = pytest.importorskip("beartype")`.

So I think the whole test should be:
```python
def test_import_hook_function_beartype(importhook_tempdir):
    beartype = pytest.importorskip("beartype")
    _test_import_hook(importhook_tempdir, beartype.beartype)
```

Reply:
Yep, I actually just made it a fixture (though it forces me to use not the standard module names). And so that ruff does not complain, I am forced to write a bit more verbosely. I prefer to do this:
```
def test(..., beartype_or_skip):
    _test_import_hook(..., beartype_or_skip.beartype)
```
over
```
def test(..., beartype_or_skip):
      _test_import_hook(..., beartype.beartype) # noqa: blahblah
```.

Though maybe you have a different opinion?

Reply:
No strong feelings; this seems fine to me.